enterprise/frontend/src/embedding-sdk/store/reducer.ts

```diff
@@ -108,69 +108,52 @@ const initialState: SdkState = {
 };
 
 export const sdk = createReducer(initialState, builder => {
-  builder.addCase(refreshTokenAsync.pending, state => ({
-    ...state,
-    token: { ...state.token, loading: true },
-  }));
-
-  builder.addCase(refreshTokenAsync.fulfilled, (state, action) => ({
-    ...state,
-    token: {
-      ...state.token,
-      token: action.payload,
-      error: null,
-      loading: false,
-    },
-  }));
-
-  builder.addCase(refreshTokenAsync.rejected, (state, action) => ({
-    ...state,
-    isLoggedIn: false,
-    token: {
-      ...state.token,
-      token: null,
-      error: action.error,
-      loading: false,
-    },
-  }));
-
-  builder.addCase(setLoginStatus, (state, action) => ({
-    ...state,
-    loginStatus: action.payload,
-  }));
-
-  builder.addCase(setLoaderComponent, (state, action) => ({
-    ...state,
-    loaderComponent: action.payload,
-  }));
+  builder.addCase(refreshTokenAsync.pending, state => {
+    state.token.loading = true;
+  });
+
+  builder.addCase(refreshTokenAsync.fulfilled, (state, action) => {
+    state.token.loading = false;
+    state.token.token = action.payload;
+    state.token.error = null;
+  });
+
+  builder.addCase(refreshTokenAsync.rejected, (state, action) => {
+    state.token.token = null;
+    state.token.error = action.error;
+    state.token.loading = false;
+  });
+
+  builder.addCase(setLoginStatus, (state, action) => {
+    state.loginStatus = action.payload;
+  });
+
+  builder.addCase(setLoaderComponent, (state, action) => {
+    state.loaderComponent = action.payload;
+  });
 
   builder.addCase(setPlugins, (state, action) => ({
     ...state,
     plugins: action.payload,
   }));
```

Comment:
If I migrate `setPlugins` to use `state.plugins = action.payload`, Immer freaks out with type errors: `TS2321: Excessive stack depth comparing types SdkPluginsConfig | null and WritableDraft<SdkPluginsConfig> | null`.